src/components/screens/ValidateRecoveryPhraseScreen.tsx

```diff
@@ -0,0 +1,95 @@
+import { NativeStackScreenProps } from "@react-navigation/native-stack";
+import { OnboardLayout } from "components/layout/OnboardLayout";
+import Icon from "components/sds/Icon";
+import { Input } from "components/sds/Input";
+import { AUTH_STACK_ROUTES, AuthStackParamList } from "config/routes";
+import { useAuthenticationStore } from "ducks/auth";
+import useAppTranslation from "hooks/useAppTranslation";
+import { useWordSelection } from "hooks/useWordSelection";
+import React, { useCallback, useMemo, useState } from "react";
+import { InteractionManager } from "react-native";
+
+type ValidateRecoveryPhraseScreenProps = NativeStackScreenProps<
+  AuthStackParamList,
+  typeof AUTH_STACK_ROUTES.VALIDATE_RECOVERY_PHRASE_SCREEN
+>;
+
+const DELAY_MS = 500;
+
+export const ValidateRecoveryPhraseScreen: React.FC<
+  ValidateRecoveryPhraseScreenProps
+> = ({ route }) => {
+  const { password, recoveryPhrase } = route.params;
+  const [currentWord, setCurrentWord] = useState("");
+  const [error, setError] = useState<string | undefined>();
+  const [isLoading, setIsLoading] = useState(false);
+  const [isSigningUp, setIsSigningUp] = useState(false);
+  const [currentIndex, setCurrentIndex] = useState(0);
+
+  const { signUp } = useAuthenticationStore();
+  const { t } = useAppTranslation();
+
+  const { words, selectedIndices } = useWordSelection(recoveryPhrase);
+  const currentWordIndex = selectedIndices[currentIndex];
+  const canContinue = useMemo(
+    () =>
+      currentWord.trim().toLowerCase() ===
+      words[currentWordIndex].toLowerCase(),
+    [currentWord, currentWordIndex, words],
+  );
+
+  const handleContinue = useCallback(() => {
+    if (!canContinue) {
+      setError(t("validateRecoveryPhraseScreen.errorText"));
+      return;
+    }
+
+    setIsLoading(true);
+
+    setTimeout(() => {
+      if (currentIndex < 2) {
+        setCurrentIndex(currentIndex + 1);
+        setCurrentWord("");
+        setError(undefined);
+      } else {
+        setIsSigningUp(true);
+        InteractionManager.runAfterInteractions(() => {
+          signUp({
+            password,
+            mnemonicPhrase: recoveryPhrase,
+          });
+        });
+      }
+      setIsLoading(false);
+    }, DELAY_MS);
```

Comment:
are we adding this timeout to  improve the UX a little bit? Like giving the user a sense of "data is being processed"
if so, should we also add this delay for the error case to keep the UX consistent?